src/simplification/Utils.ts

```diff
@@ -0,0 +1,17 @@
+import { CommonModel } from '../models/CommonModel';
+
+/**
+ * check if CommonModel is a separate model or a simple model.
+ */
+export function isModelObject(model: CommonModel) : boolean {
+  // This check should be done instead, needs a refactor to allow it though:
+  // this.extend !== undefined || this.properties !== undefined
+  if (model.type !== undefined) {
+    // If all possible JSON types are defined, don't split it even if it does contain object.
+    if (Array.isArray(model.type) && model.type.length === 6) {
```

Comment:
Maybe this is time ti change that? 😄

Reply:
#59 addresses this if merged, with the current implementation this is not possible because of the recursive nature where each CommonModel does not have access to those properties yet 😄